data/src/main/java/com/microsoft/azure/kusto/data/ClientDetails.java

```diff
@@ -1,63 +1,55 @@
 package com.microsoft.azure.kusto.data;
 
-import org.apache.commons.lang3.StringUtils;
-import org.apache.commons.lang3.concurrent.ConcurrentException;
-import org.apache.commons.lang3.concurrent.ConcurrentInitializer;
-import org.apache.commons.lang3.concurrent.LazyInitializer;
-import org.apache.commons.lang3.tuple.Pair;
+import com.azure.core.util.CoreUtils;
 import reactor.util.annotation.Nullable;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
 public class ClientDetails {
 
     public static final String NONE = "[none]";
-    private static ConcurrentInitializer<String> defaultApplication = new LazyInitializer() {
-        @Override
-        protected String initialize() {
-            return UriUtils.stripFileNameFromCommandLine(System.getProperty("sun.java.command"));
-        }
-    };
-    private static ConcurrentInitializer<String> defaultUser = new LazyInitializer() {
-        @Override
-        protected String initialize() {
-            String user = System.getProperty("user.name");
-            if (StringUtils.isBlank(user)) {
-                user = System.getenv("USERNAME");
-                String domain = System.getenv("USERDOMAIN");
-                if (StringUtils.isNotBlank(domain) && StringUtils.isNotBlank(user)) {
-                    user = domain + "\\" + user;
-                }
-            }
-            return StringUtils.isNotBlank(user) ? user : NONE;
-        }
-    };
-    private static ConcurrentInitializer<String> defaultVersion = new LazyInitializer() {
-        @Override
-        protected String initialize() {
-            return formatHeader(Arrays.asList(
-                    Pair.of("Kusto.Java.Client", Utils.getPackageVersion()),
-                    Pair.of(String.format("Runtime.%s", escapeField(getRuntime())), getJavaVersion())));
-        }
-    };
-    private String applicationForTracing;
-    private String userNameForTracing;
-    private String appendedClientVersionForTracing;
+
+    private static final ConcurrentHashMap<String, String> defaultValues = new ConcurrentHashMap<>();
+    public static final String DEFAULT_APPLICATION = "defaultApplication";
```

Comment:
Cool idea to use ConcurrentHasmap's computeifabsent and then access these as keys.

But maybe instead of strings, it should be an enum?